### ISSUE TITLE
GitHub Latest multiple buttons

### ISSUE THREAD

**Issue by jerone** [Ede123/userscripts#4]
On some pages I get multiple buttons:
![](https://user-images.githubusercontent.com/55841/30203915-08a75a26-9484-11e7-966a-37a300feebc6.png)

Reproducible on the "Insides" > "Graphs" > "Traffic".

**Comment by Ede123** [comment]
Hmm, I'm not able to reproduce this issue.

Do you have more specific steps on how to reproduce? Might it have been a temporary glitch?

**Comment by jerone** [comment]
It's a little hard to reproduce consistently, but what did it for me was:
1. Open Insides.
2. Open Graphs.
3. Open Traffic.
4. Go into one referring site.
5. Go back and go into another referring site.
6. Go back and return to the referring site in step 4.

It looks like it happens when you have a "cached" route in pjax (I think).

**Comment by Ede123** [comment]
Ah, ok, now I see it... the "Go into one referring site." part triggers it.
(I think I tried that before but maybe I was doing something differently)

**Comment by jerone** [comment]
A solution might be to append an unique id to the button and check if it doesn't exists, like [I do](https://github.com/jerone/UserScripts/blob/master/Github_Commit_Whitespace/Github_Commit_Whitespace.user.js#L26-L29) with [my](https://github.com/jerone/UserScripts/blob/master/Github_Commit_Diff/Github_Commit_Diff.user.js#L28-L31) scripts.

**Comment by Ede123** [comment]
Fixed in 319b963d13bdd5660412930215e388847d2baa09.

I went with the button ID as it was simple yet efficient.

Alternatively the event data of the `pjax.success` event could be parsed. It contains for example the `currentTarget` attribute which could be stored / compared to see if it changed:

```
pjax:success { target: <div#top-domains.column.one-half.top-domains>, isTrusted: false, currentTarget: HTMLDocument → traffic, eventPhase: 3, bubbles: true, cancelable: true, defaultPrevented: false, composed: false, timeStamp: 9472.072493909063, cancelBubble: false, originalTarget: <div#top-domains.column.one-half.top-domains> }
```
